### PR TITLE
Fix docblock annotation for PublisherInterface message

### LINKED ISSUE #30190: [Issue] Fix docblock annotation for PublisherInterface message
This issue is automatically created based on existing pull request: magento/magento2#30180: Fix docblock annotation for PublisherInterface message

---------

### Description (*)

The message on the publisher interface's `publish` method currently has a docblock annotation of `array|object`. However, since this type can be anything other than array or object as long as it's defined in the `communication.xml` this docblock annotation creates confusion. The docblock annotation is therefore wrong. This PR fixes that.

### Manual testing scenarios (*)

Since this is a doc change, I'm not sure if there's any manual testing scenarios? Please tell me if I'm wrong.

### Contribution checklist (*)
 - [x] Pull request has a meaningful description of its purpose
 - [x] All commits are accompanied by meaningful commit messages
 - [ ] All new or changed code is covered with unit/integration tests (if applicable)
 - [ ] All automated tests passed successfully (all builds are green)

### PR DESCRIPTION
**Pull request by kolaente**
### Description (*)

The message on the publisher interface's `publish` method currently has a docblock annotation of `array|object`. However, since this type can be anything other than array or object as long as it's defined in the `communication.xml` this docblock annotation creates confusion. The docblock annotation is therefore wrong. This PR fixes that.

### Manual testing scenarios (*)

Since this is a doc change, I'm not sure if there's any manual testing scenarios? Please tell me if I'm wrong.

### Contribution checklist (*)
 - [x] Pull request has a meaningful description of its purpose
 - [x] All commits are accompanied by meaningful commit messages
 - [ ] All new or changed code is covered with unit/integration tests (if applicable)
 - [ ] All automated tests passed successfully (all builds are green)

### Resolved issues:
1. [x] resolves magento/magento2#30190: Fix docblock annotation for PublisherInterface message

### CODE REVIEW

**Review by rogyar — CHANGES_REQUESTED** [review]
Hi @kolaente. Please, sign Adobe CLA. So we can proceed with processing the PR.

Thank you

**Comment by kolaente** [comment]
@rogyar Done!

**Comment by sidolov** [comment]
@magento create issue

**Review by rogyar — DISMISSED** [review]

**Comment by magento-engcom-team** [comment]
Hi @rogyar, thank you for the review.
[ENGCOM-8277](https://jira.corp.magento.com/browse/ENGCOM-8277) has been created to process this Pull Request

**Comment by engcom-Alfa** [comment]
QA not applicable

**Comment by engcom-Foxtrot** [comment]
@magento run all tests

**Comment by sidolov** [comment]
Static tests are failed.

**Comment by engcom-Foxtrot** [comment]
@kolaente param type should be specified. Please use "mixed".

File: lib/internal/Magento/Framework/MessageQueue/PublisherInterface.php [hunk 1]
```diff
@@ -18,7 +18,7 @@ interface PublisherInterface
      * Publishes a message to a specific queue or exchange.
      *
      * @param string $topicName
-     * @param array|object $data
+     * @param $data
```
Comment (engcom-Foxtrot) [inline lib/internal/Magento/Framework/MessageQueue/PublisherInterface.php:21]:
```suggestion
     * @param mixed $data
```

Comment (engcom-Foxtrot) [inline lib/internal/Magento/Framework/MessageQueue/PublisherInterface.php]:
```suggestion
     * @param mixed $data
```

**Review by engcom-Foxtrot — CHANGES_REQUESTED** [review]

File: lib/internal/Magento/Framework/MessageQueue/PublisherInterface.php (hunk 1 above)
Comment (kolaente) [reply lib/internal/Magento/Framework/MessageQueue/PublisherInterface.php]: Done!

**Comment by kolaente** [comment]
@engcom-Foxtrot @sidolov Should be fixed now.

**Comment by engcom-Foxtrot** [comment]
@magento run all tests

**Review by engcom-Foxtrot — APPROVED** [review]

**Comment by engcom-Foxtrot** [comment]
@sidolov fixed.

**Pull request merged**